xray/conventions.py

```diff
@@ -871,6 +874,9 @@ def decode_cf(obj, concat_characters=True, mask_and_scale=True,
     vars, attrs, coord_names = decode_cf_variables(
         vars, attrs, concat_characters, mask_and_scale, decode_times,
         decode_coords)
+    if drop_variables is not None:
```

Comment:
could you move this into `decode_cf_variables` instead? probably could put this logic in the loop that constructs `new_vars`